mantidproject/mantid
# Changes to various unit and system test files to clean up temp files

**Description of work.**

Remove files written to the temporary directory when the unit tests and system tests are run. This was causing problems with the build servers running out of space.

One of these problems was caused by a system test (ILLPowderEfficiencyClosureTest.py) having a tearDown method that cleaned up the file. The system test runner never calls tearDown. So I renamed this to cleanup (which is called) - I have also made a similar change in 7 other system test units which weren't leaving files behind but had a tearDown method that was never called.

**To test:**

Run the entire system test and unit test set (Python and C++) and check:
a) if there are any files left in the temp directory (which is C:\Users\<user id>\AppData\Local\Temp on Windows and I think simply \tmp on Linux). I have listed the files that were left behind before the changes on the issue
b) the tests still work

Alternatively you could just run the individual tests listed on the issue. This is more manual work but perhaps less waiting around.

Fixes #27058.

*This does not require release notes* because it's a maintenance change that only impacts the build process

---

#### Reviewer ####

Please comment on the following ([full description](http://developer.mantidproject.org/ReviewingAPullRequest.html)):

##### Code Review #####

- Is the code of an acceptable quality?
- Does the code conform to the [coding standards](http://developer.mantidproject.org/Standards/)?
- Are the unit tests small and test the class in isolation?
- If there are changes in the release notes then do they describe the changes appropriately?

##### Functional Tests #####

- Do changes function as described? Add comments below that describe the tests performed?
- Do the changes handle unexpected situations, e.g. bad input?
- Has the relevant (user and developer) documentation been added/updated?

Does everything look good? Mark the review as **Approve**. A member of `@mantidproject/gatekeepers` will take care of it.
